### ISSUE TITLE
[4.x]: Drag and drop not working on Neo child blocks

### ISSUE THREAD

**Issue by chrismlusk** [craftcms/cms#12896]
### Description

When attempting to grab a Neo child block and reorder it, the entire parent block is grabbed instead.

[An issue was opened in the Neo repo](https://github.com/spicywebau/craft-neo/issues/719), but it was closed with an assumption that it is a Craft bug.

However, I tested with Matrix fields, and I can drag and drop child blocks as expected. Even a Matrix field within a Neo block works, too.

So, it might not be a Craft bug, but I wanted to open an issue just in case.

### Craft CMS version

4.4.1

### Installed plugins and versions

- Neo: 3.7.1

**Comment by ttempleton** [comment]
Extra context: I had assumed it's a Craft bug because it worked properly with Craft 4.3. I've just done some quick testing and found that it was working as late as 4.4.0-beta.4 and specifically seems to have been introduced with https://github.com/craftcms/cms/commit/a6eeabfca395517047e0aaece0a94edfaf381450.

**Comment by brandonkelly** [comment]
3.8.3 and 4.4.3 are out with the fix!

**Comment by engram-design** [comment]
Just FYI, this has broken Navigation (just for Craft 3) which uses a manual structure setup. Refer to https://github.com/verbb/navigation/issues/348

What I don't _quite_ understand is the thinking here, because according to [StructureDrag.js](https://github.com/craftcms/cms/blob/v3/src/web/assets/cp/src/js/StructureDrag.js#L24) you can have multiple selectors for the handles of the drag trigger:

```js
this.base($items, {
    handle: '.element:first, .move:first',
    helper: this.getHelper.bind(this),
});
```

Because this has been changed to use only `first()` on this selector, it means only `.element:first` works, which is exactly what we're seeing. Removing the change made in https://github.com/craftcms/cms/pull/12911/commits/66ae85b980000f2e35f563a71e3e04fe2c034985 fixes our issue.

To put this visually, you can see the difference when calling:

```js
console.log($(this.settings.handle, item));
console.log($(this.settings.handle, item).first());
```

<img width="682" alt="image" src="https://github.com/craftcms/cms/assets/1221575/d00ca069-65ba-469d-80cc-38a722ccd7fa">

So the `.move:first` selector never actually gets registered with the `BaseDrag` class, and therefore the `StructureDrag` that we use. I'm looking into how this still seems to work for Sections which use the structure layout, so it's certainly a Navigation-specific thing, but I'm sure others might be using the `StructureDrag` component.

**Comment by i-just** [comment]
@engram-design I have a few ideas on this; we’ll discuss it locally, and I’ll get back to you.

**Comment by brandonkelly** [comment]
@engram-design We released Craft 3.8.12 today with a fix for that 🎉

**Comment by engram-design** [comment]
Seems to have sorted it, thanks!
